src/ansible_navigator/configuration_subsystem/navigator_post_processor.py

```diff
@@ -46,13 +49,64 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+class VolumeMountOption(Enum):
+    """Options that can be tagged on to the end of volume mounts.
+
+    Usually these are used for things like selinux relabeling, but there are
+    some other valid options as well, which can and should be added here as
+    needed. See ``man podman-run`` and ``man docker-run`` for valid choices and
+    keep in mind that we support both runtimes.
+    """
+
+    # Relabel as private
+    Z = "Z"
+
+    # Relabel as shared.
+    z = "z"  # pylint: disable=invalid-name
+
+
+@dataclass
+class VolumeMount:
+    """Describes EE volume mounts."""
+
+    # The name of the config option requiring this volume mount.
+    calling_option: str
+
+    # The source path of the volume mount.
+    src: str
+
+    # The destination path in the container for the volume mount.
+    dest: str
+
+    # Options for the bind mount.
+    options: Optional[List[VolumeMountOption]] = None
+
+    def exists(self) -> bool:
+        """Does the volmount source exist?"""
+        return Path(self.src).exists()
+
+    def to_string(self) -> str:
+        """Render the volume mount in a way that (docker|podman) understands."""
+        out = f"{self.src}:{self.dest}"
+        if self.options:
+            joined_opts = ",".join([o.value for o in self.options])
```

Comment:
I don't see why a generator expression isn't enough.

```suggestion
            joined_opts = ",".join(o.value for o in self.options)
```

Reply:
Ask the linter that :)

Reply:
Not that my change fixed anything - I was just trying something.

Reply:
I _think_ what's happening is that the linter thinks it's not iterable because it can be `None`. It's not smart enough to know that I'm actually checking that above it.

Reply:
I would also check its type in the if condition and thus should be enough for mypy to determine that its must be a List. I used something similar in other cases.

Reply:
@relrod

> I _think_ what's happening is that the linter thinks it's not iterable because it can be `None`. It's not smart enough to know that I'm actually checking that above it.

Could you be more specific about which linter and what the error message was originally?

If Sorin is right and it indeed was `mypy`, one of the ways to make a hint to it, is to add `assert self.options is not None` and if it was `Optional[List]` before, then after than line `mypy` will *know* that it's just a `List`. Another way is to use `typing.cast()`.

Reply:
Oh, so looked into Zuul output and it's 
```console
src/ansible_navigator/configuration_subsystem/navigator_post_processor.py:92:53: E1133: Non-iterable value self.options is used in an iterating context (not-an-iterable)
```
Not sure if an assert would help here.